Pyramid.c

```diff
@@ -0,0 +1,130 @@
+#include <stdio.h>
+#include <string.h>
+
+#define MAX_DIGITS 80
+
+struct BigInt {
+int digits_count;
+unsigned int the_int[MAX_DIGITS];
+};
+
+
+int strtobig_int(const char *str, int len, struct BigInt *big_int);
+void print_big_int(const struct BigInt *big_int);
+void multiply(const struct BigInt *big_int, int factor, struct BigInt *big_result);
+void divide(const struct BigInt *big_int, int divisor, struct BigInt *big_result);
+void copy_big_int(const struct BigInt *from, struct BigInt *to);
+
+
+int main(int argc, char *argv[])
+{
+char input[MAX_DIGITS + 1];
+printf("Geben sie eine nummer ein: ");
+scanf("%s",input);
+int len = strlen(input);
+if (len > MAX_DIGITS) {
+return 0;
+}
+for (int i = 0; i < len; i++) {
+if (!(input[i] > 47 && input[i] < 58)) {
+printf("Error code -1\n");
+return 0;
+}
+}
+struct BigInt big_int;
+struct BigInt big_result;
+big_int.digits_count = strtobig_int(input, len, &big_int);
```

Comment:
The next line should not be necessary. Setting the digits_count of the BigInt is responsibility of multiply, divide, etc.